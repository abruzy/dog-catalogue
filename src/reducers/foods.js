const INITIAL_STATE = { foods_recipies: [], food_item: {} };

const foods = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case 'FETCH RECIPIES':
      return { ...state, foods_recipies: action.payload };
    case 'FETCH RECIPE':
      return { ...state, food_item: action.payload };
    case 'FILTER BY CATEGORY':
      return { ...state, foods_recipies: action.payload.data.meals };
    case 'RESET_DATA':
      return { ...state, foods_recipies: [] };
    default:
      return state;
  }
};

export default foods;
