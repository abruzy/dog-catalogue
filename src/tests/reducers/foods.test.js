import foods from '../../reducers/foods';

describe('foods reducer', () => {
  it('should return the initial state', () => {
    expect(foods(undefined, {})).toMatchSnapshot();
  });

  it('should handle FETCH RECIPIES', () => {
    expect(foods([], {
      type: 'FETCH RECIPIES',
    })).toMatchSnapshot();
  });

  it('should handle FETCH RECIPE', () => {
    expect(foods({}, {
      type: 'FETCH RECIPE',
    })).toMatchSnapshot();
  });

  it('should handle RESET DATA', () => {
    expect(foods([], {
      type: 'RESET_DATA',
    })).toMatchSnapshot();
  });
});
