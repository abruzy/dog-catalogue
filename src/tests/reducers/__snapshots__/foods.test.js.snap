// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`foods reducer should handle FETCH RECIPE 1`] = `
Object {
  "food_item": undefined,
}
`;

exports[`foods reducer should handle FETCH RECIPIES 1`] = `
Object {
  "foods_recipies": undefined,
}
`;

exports[`foods reducer should handle RESET DATA 1`] = `
Object {
  "foods_recipies": Array [],
}
`;

exports[`foods reducer should return the initial state 1`] = `
Object {
  "food_item": Object {},
  "foods_recipies": Array [],
}
`;
